File: test-suite-python/src/test/python/micronaut/docs/aws/alexa/handlers/HelloWorldIntentHandler.py
# tag::clazz[]
from com.amazon.ask.dispatcher.request.handler import HandlerInput, RequestHandler
from com.amazon.ask.model import Response
from com.amazon.ask.request import Predicates
from jakarta.inject import Singleton
from java.util import Optional


@Singleton  # <1>
class HelloWorldIntentHandler(RequestHandler):

    def canHandle(self, input: HandlerInput) -> bool:
        return input.matches(Predicates.intentName("HelloWorldIntent"))

    def handle(self, input: HandlerInput) -> Optional[Response]:
        speech_text = "Hello world"
        return (input.getResponseBuilder()
                .withSpeech(speech_text)
                .withSimpleCard("HelloWorld", speech_text)
                .build())
# end::clazz[]

File: test-suite/src/test/java/io/micronaut/docs/aws/alexa/handlers/HelloWorldIntentHandler.java
package io.micronaut.docs.aws.alexa.handlers;

//tag::clazz[]
import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.Response;
import com.amazon.ask.request.Predicates;
import jakarta.inject.Singleton;

import java.util.Optional;

@Singleton // <1>
public class HelloWorldIntentHandler implements RequestHandler {

    @Override
    public boolean canHandle(HandlerInput input) {
        return input.matches(Predicates.intentName("HelloWorldIntent"));
    }

    @Override
    public Optional<Response> handle(HandlerInput input) {
        String speechText = "Hello world";
        return input.getResponseBuilder()
                .withSpeech(speechText)
                .withSimpleCard("HelloWorld", speechText)
                .build();
    }

}
//end::clazz[]
